utils/analysis/filter.go

```diff
@@ -0,0 +1,75 @@
+package analysis
+
+import (
+	"strconv"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+)
+
+// FilterCurrentRolloutAnalysisRuns returns analysisRuns that match the analysisRuns listed in the rollout status
+func FilterCurrentRolloutAnalysisRuns(analysisRuns []*v1alpha1.AnalysisRun, r *v1alpha1.Rollout) ([]*v1alpha1.AnalysisRun, []*v1alpha1.AnalysisRun) {
+	return filterAnalysisRuns(analysisRuns, func(ar *v1alpha1.AnalysisRun) bool {
+		if ar.Name == r.Status.Canary.CurrentStepAnalysisRun {
+			return true
+		}
+		if ar.Name == r.Status.Canary.CurrentParellelAnalysisRun {
+			return true
+		}
+		return false
+	})
+}
+
+// FilterAnalysisRunsByRolloutType returns a list of analysisRuns that have the rollout-type of the typeFilter
+func FilterAnalysisRunsByRolloutType(analysisRuns []*v1alpha1.AnalysisRun, typeFilter string) ([]*v1alpha1.AnalysisRun, []*v1alpha1.AnalysisRun) {
```

Comment:
Based on the caller of this function, seems like we only care about the matching case, so I feel like we don't need to return the second return value containing the non-matching runs.

Reply:
Agreed